### PR TITLE
Refactor/build and tests

### PR DESCRIPTION
**Pull request by ritch**
/to @jtary

### CODE REVIEW

**Comment by jtary** [comment]
LGTM

**Comment by ritch** [comment]
test please

**Comment by ritch** [comment]
test please

File: test/karma.conf.js [hunk 1]
```diff
@@ -1,33 +1,29 @@
...
-
-    // list of files to exclude
-    exclude: [
-
-    ],
-
-
     // test results reporter to use
     // possible values: 'dots', 'progress', 'junit', 'growl', 'coverage'
     reporters: ['progress'],
```
Comment (rmg) [inline test/karma.conf.js]: something like `reporters: [process.env.CI ? 'junit' : 'progress'],` would help CI a lot

Comment (rmg) [inline test/karma.conf.js]: Even better, throw in a coverage reporter as shown in https://github.com/karma-runner/karma-coverage

Comment (rmg) [inline test/karma.conf.js]: Just make sure the coverage reporter is configured to use cobertura format output to a file called `cobertura-coverage.xml`.

**Comment by ritch** [comment]
test please

**Comment by ritch** [comment]
test please

**Pull request merged**